dot/parachain/backing/candidate_backing.go

```diff
@@ -1,121 +1,664 @@
+// Copyright 2023 ChainSafe Systems (ON)
+// SPDX-License-Identifier: LGPL-3.0-only
+
 package backing
 
 import (
 	"context"
+	"errors"
+	"fmt"
 
 	parachaintypes "github.com/ChainSafe/gossamer/dot/parachain/types"
 	"github.com/ChainSafe/gossamer/internal/log"
 	"github.com/ChainSafe/gossamer/lib/common"
+	"github.com/ChainSafe/gossamer/pkg/scale"
 )
 
 var logger = log.NewFromGlobal(log.AddContext("pkg", "parachain-candidate-backing"))
 
+var (
+	ErrRejectedByProspectiveParachains = errors.New("candidate rejected by prospective parachains subsystem")
+	ErrValidationFailed                = errors.New("validation failed")
+	ErrInvalidErasureRoot              = errors.New("erasure root doesn't match the announced by the candidate receipt")
+)
+
 type CandidateBacking struct {
 	SubSystemToOverseer chan<- any
 	OverseerToSubSystem <-chan any
+	perRelayParent      map[common.Hash]perRelayParentState
+	perCandidate        map[parachaintypes.CandidateHash]perCandidateState
+}
+
+type perCandidateState struct {
+	persistedValidationData parachaintypes.PersistedValidationData
+	SecondedLocally         bool
+	ParaID                  parachaintypes.ParaID
+	RelayParent             common.Hash
+}
```

Comment:
Add comment for these properties. It would be useful to use when we read the code later.